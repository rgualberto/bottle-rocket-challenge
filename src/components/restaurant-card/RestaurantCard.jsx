import React from 'react';
import PropTypes from 'prop-types';
import './restaurant-card.scss';

const RestaurantCard = props => {
  const {
    Name,
    Category,
    ImageURL,
    onClick
  } = props;

  return (
    <button
      className="restaurant-card"
      onClick={onClick}
    >
      <img
        className="restaurant-card__image"
        src={ImageURL}
      />

      <div className="restaurant-card__info">
        <h2>{Name}</h2>
        <span>{Category}</span>
      </div>
    </button>
  );
}

RestaurantCard.propTypes = {
  Name: PropTypes.string.isRequired,
  Category: PropTypes.string.isRequired,
  ImageURL: PropTypes.string.isRequired,

  onClick: PropTypes.func.isRequired
};

export default RestaurantCard;
