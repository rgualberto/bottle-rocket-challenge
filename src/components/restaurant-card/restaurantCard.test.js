import React from 'react';
import renderer from 'react-test-renderer';
import RestaurantCard from './RestaurantCard.jsx';

test('RestaurantCard snapshot', () => {
  const component = renderer.create(
    <RestaurantCard
      Name="Test Restaurant"
      Category="Tasty"
      ImageURL=""
      onClick={() => undefined}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
