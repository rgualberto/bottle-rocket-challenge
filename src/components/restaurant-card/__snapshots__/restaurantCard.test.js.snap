// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard snapshot 1`] = `
<button
  className="restaurant-card"
  onClick={[Function]}
>
  <img
    className="restaurant-card__image"
    src=""
  />
  <div
    className="restaurant-card__info-wrap"
  >
    <div
      className="restaurant-card__info"
    >
      <h2>
        Test Restaurant
      </h2>
      <span>
        Tasty
      </span>
    </div>
  </div>
</button>
`;
